nilearn/masking.py

```diff
@@ -607,32 +612,44 @@ def compute_brain_mask(target_img, threshold=.5, connected=True,
         Controls the amount of verbosity: higher numbers give
         more messages
 
+    mask_type : {'whole-brain', 'gm', 'wm'}, optional
+        Type of mask to be computed. Default = 'whole-brain'
+
     Returns
     -------
     mask : nibabel.Nifti1Image
         The whole-brain mask (3D image)
     """
     if verbose > 0:
-        print("Template mask computation")
+        msg_to_print = "Template " + mask_type + " mask computation"
+        print(msg_to_print)
 
     target_img = _utils.check_niimg(target_img)
 
-    from .datasets import load_mni152_brain_mask
-    template = load_mni152_brain_mask()
+    if mask_type == 'whole-brain':
+        template = load_mni152_brain_mask()
+    elif mask_type == 'gm':
+        template = load_mni152_gm_mask()
+    elif mask_type == 'wm':
+        template = load_mni152_wm_mask()
+    else:
+        raise ValueError("Unknown mask type {}".format(mask_type))
```

Comment:
@NicolasGensollen, Codecov is not covering this line in the tests. Perhaps, something like this would solve the problem:

``` python
if mask_type == 'whole-brain':
    template = load_mni152_brain_mask()
elif mask_type == 'gm':
    template = load_mni152_gm_mask()
else: 
    assert mask_type == 'wm'
    template = load_mni152_wm_mask()
```

Reply:
Why not simply add a line in `test_compute_brain_mask()`?

```python
with pytest.raises(ValueError,
                   match='Unknown mask type foo.'):
      compute_brain_mask(img2, verbose=1, mask_type='foo')
```